src/protocol/ws/non_wasm.rs

```diff
@@ -50,57 +51,89 @@ use tokio_tungstenite::WebSocketStream;
 #[async_trait]
 impl Connection for Client {
     type Request = (i64, Message);
-    type Response = Value;
+    type Response = Result<DbResponse>;
     type Client = Self;
 
     fn new(method: Method) -> Self {
         Self { id: 0, method }
     }
 
-    async fn send<A>(
+    async fn send(
         &mut self,
         router: &Router<Self>,
-        msg: A,
-    ) -> Result<oneshot::Receiver<Result<Value>>>
-    where
-        A: Serialize + Send,
-    {
+        Param(first, second): Param,
+    ) -> Result<oneshot::Receiver<Self::Response>> {
         self.id = match self.method {
             // We don't want to use up IDs on pings
             Method::Ping => 0,
             _ => router.next_id(),
         };
-        let payload = serde_json::to_string(&Request {
-            id: self.id,
-            method: self.method,
-            params: Some(msg),
-        })
-        .unwrap();
+        let id = Value::from(self.id);
+        let method = match json!(self.method) {
```

Comment:
Out of interest, what's the point in converting the `method` to JSON first?

Reply:
It's a convoluted way of converting the `Method` enum to a string :laughing: Yeah, I hate it too :rofl:  I didn't want to type out the method names by hand. I will bite the bullet and add `Method::as_str` that converts the enum to a `&'static str`.